drivers/SmartThings/philips-hue/src/init.lua

```diff
@@ -641,31 +667,27 @@ cosock.spawn(function()
 
     local msg_device, driver = msg.device, msg.driver
     if msg.type == StrayDeviceMessageTypes.FoundBridge then
-      local dnis_to_remove = {}
       local bridge_ip = msg_device:get_field(Fields.IPV4)
-      local api_instance = HueApi.new_bridge_manager("https://" .. bridge_ip, msg_device:get_field(Fields.API_KEY))
-      Discovery.search_bridge_for_supported_devices(driver, api_instance, function(hue_driver, svc_info, device_data)
-        if not (svc_info.rid and svc_info.rtype and svc_info.rtype == "light") then return end
-
-        for light_dni, light_device in pairs(stray_lights) do
-          local matching_v1_id = light_device.data and light_device.data.bulbId and
-              light_device.data.bulbId == device_data.id_v1:gsub("/lights/", "")
-          local matching_uuid = light_device.device_network_id == svc_info.rid or
-              light_device.device_network_id == svc_info.rid
-
-          if matching_v1_id or matching_uuid then
-            log.trace("Found Bridge for stray light ", light_device.label, ", re-adding")
-            table.insert(dnis_to_remove, light_dni)
-            _initialize(hue_driver, light_device, nil, nil, msg_device.id)
-          end
-        end
-      end)
+      local api_instance =
+          msg_device:get_field(Fields.BRIDGE_API) or
+          HueApi.new_bridge_manager("https://" .. bridge_ip, msg_device:get_field(Fields.API_KEY))
 
-      for _, dni in ipairs(dnis_to_remove) do
-        stray_lights[dni] = nil
-      end
+      found_bridges[msg_device.id] = msg.device
+      process_strays(driver, api_instance, stray_lights, msg_device.id)
     elseif msg.type == StrayDeviceMessageTypes.NewStrayLight then
       stray_lights[msg_device.device_network_id] = msg_device
+
+      local maybe_bridge_id =
+      msg_device.parent_device_id or msg_device:get_field(Fields.PARENT_DEVICE_ID)
+  local maybe_bridge = found_bridges[maybe_bridge_id]
+
+  if maybe_bridge ~= nil then
+    local bridge_ip = maybe_bridge:get_field(Fields.IPV4)
+    local api_instance =
+        maybe_bridge:get_field(Fields.BRIDGE_API) or
+        HueApi.new_bridge_manager("https://" .. bridge_ip, msg_device:get_field(Fields.API_KEY))
+    process_strays(driver, api_instance, stray_lights, maybe_bridge.id)
+  end
```

Comment:
nit: indent is confusing here. I think this is a part of the` elseif msg.type == StrayDeviceMessageTypes.NewStrayLight` branch.

Reply:
Oof yeah the formatting got messed up here. Thanks.